File: mobile_app/lib/domain/mode_five.dart
import 'package:rvi_analyzer/domain/default_configuration.dart';
import 'package:rvi_analyzer/domain/session_result.dart';

class ModeFive {
  final String createdBy;
  final DefaultConfiguration defaultConfigurations;
  final SessionConfigurationModeFive sessionConfigurationModeFive;
  final SessionResult results;
  final String status;

  ModeFive(
      {required this.createdBy,
      required this.defaultConfigurations,
      required this.sessionConfigurationModeFive,
      required this.results,
      required this.status});

  Map toJson() => {
        'createdBy': createdBy,
        'defaultConfigurations': defaultConfigurations.toJson(),
        'sessionConfigurationModeFive': sessionConfigurationModeFive.toJson(),
        'results': results,
        'status': status,
      };
}

class SessionConfigurationModeFive {
  final String fixedVoltage;
  final String maxCurrent;
  final String timeDuration;

  SessionConfigurationModeFive(
      {required this.fixedVoltage,
      required this.maxCurrent,
      required this.timeDuration});

  Map toJson() => {
        'fixedVoltage': fixedVoltage,
        'maxCurrent': maxCurrent,
        'timeDuration': timeDuration
      };
}

File: rvi-analyzer-server/src/main/java/com/rvi/analyzer/rvianalyzerserver/entiy/ModeFive.java
package com.rvi.analyzer.rvianalyzerserver.entiy;

import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "ModeFive")
public class ModeFive {
    @Id
    private Long _id;
    @Column
    private String createdBy;
    @Column
    private DefaultConfiguration defaultConfigurations;
    @Column
    private SessionConfigurationModeFive sessionConfigurationModeFive;
    @Column
    private SessionResult results;
    @Column
    private String status;
    @Column
    @CreatedDate
    private LocalDateTime createdDateTime;
    @Column
    @LastModifiedDate
    private LocalDateTime lastUpdatedDateTime;
}
